dts/dts.cmake

```diff
@@ -20,21 +20,29 @@ if(CONFIG_HAS_DTS)
     -include ${DTS_SOURCE}
     )
 
-  if(NOT DTC_OVERLAY_FILE)
-    # overlay file is not set, so try to use overlay dir instead.
+  set_ifndef(BOARD_OVERLAY_DIR ${APPLICATION_SOURCE_DIR})
 
-    set_ifndef(DTC_OVERLAY_DIR ${APPLICATION_SOURCE_DIR})
-
-    set(DTC_OVERLAY_FILE ${DTC_OVERLAY_DIR}/${BOARD_FAMILY}.overlay)
+  set(BOARD_OVERLAY_FILE ${DTC_OVERLAY_DIR}/${BOARD_FAMILY}.overlay)
+  if(NOT EXISTS ${BOARD_OVERLAY_FILE})
+    unset(BOARD_OVERLAY_FILE)
   endif()
 
-  if(EXISTS ${DTC_OVERLAY_FILE})
-    set(DTC_INCLUDE_FLAG_FOR_OVERLAY
-      -include ${DTC_OVERLAY_FILE}
-      )
-  else()
-    unset(DTC_INCLUDE_FLAG_FOR_OVERLAY)
-  endif()
+  # Bring in extra overlay files dropped in by the user or anyone else;
+  # make sure they are set at the end so we can override any other setting
+  file(GLOB overlay_files ${APPLICATION_BINARY_DIR}/*.overlay)
+  list(SORT overlay_files)
+  set(
+    dtc_overlay_files
```

Comment:
If we had the DTS source as the first element here we could get rid of some duplicate code I believe.

Reply:
Yep, I can fix that.